src/tests/system/tests/test_sssctl.py

```diff
@@ -684,30 +655,43 @@ def test_sssctl__check_ldap_host_object_class_in_domain(client: Client):
     assert result.rc == 0, "Config-check failed"
 
 
+@pytest.mark.importance("high")
 @pytest.mark.tools
 @pytest.mark.ticket(bz=1677994)
 @pytest.mark.topology(KnownTopology.Client)
-def test_sssctl__check_ldap_host_object_class_not_allowed_in_sssd(client: Client):
-    """
-    :title: sssctl config-check do not allow ldap_host_object_class in sssd section
+@pytest.mark.parametrize(
+    "section,option,value,expected",
+    [
+        ("domain", "services", "nss, pam", "Attribute 'services' is not allowed in section 'domain/local'"),
+        (
+            "sssd",
+            "ldap_host_object_class",
+            "ipService",
+            "Attribute 'ldap_host_object_class' is not allowed in section 'sssd'",
+        ),
+    ],
+)
+def test_sssctl__check_attribute_not_allowed_in_sssd(
+    client: Client, section: str, option: str, value: str, expected: str
+):
+    """
+    :title: sssctl config-check validates attributes in specific sections
     :setup:
-        1. Add ldap_host_object_class to sssd section
-        2. Start SSSD
+        1. Add an invalid option to a section in the configuration and start SSSD
     :steps:
-        1. Call sssctl config-check
+        1. Check the configuration using sssctl
     :expectedresults:
-        1. config-check succeed
+        1. The config check succeed with the warning in the output
     :customerscenario: True
     """
+    client.sssd.default_domain = "local"
     client.sssd.common.local()
-    client.sssd.sssd["ldap_host_object_class"] = "ipService"
+    getattr(client.sssd, section)[option] = value
     client.sssd.start(check_config=False)
```

Comment:
Actually, why is it required to start SSSD here?

Reply:
`client.sssd.start(check_config=False)` includes `self.config_apply(check_config=False)`. 
We can safely replace `client.sssd.start(check_config=False)` with `self.config_apply(check_config=False)` here but `start` is a common practice in most of the tests in this module. I personally copied the original test that has `start` method.

Reply:
I would vote to replace. At least it should be faster.